tslearn/matrix_profile.py

```diff
@@ -91,15 +97,27 @@ class MatrixProfile(TransformerMixin,
 
     References
     ----------
-    .. [1] C. M. Yeh, Y. Zhu, L. Ulanova, N.Begum et al. 
-       Matrix Profile I: All Pairs Similarity Joins for Time Series: A 
+    .. [1] C. M. Yeh, Y. Zhu, L. Ulanova, N.Begum et al.
+       Matrix Profile I: All Pairs Similarity Joins for Time Series: A
        Unifying View that Includes Motifs, Discords and Shapelets.
        ICDM 2016.
     """
 
-    def __init__(self, subsequence_length=1, scale=True):
+    def __init__(self, subsequence_length=1, algorithm=None, scale=True):
         self.subsequence_length = subsequence_length
         self.scale = scale
+        self.check_if_algorithm_exists(algorithm)
+        self.algorithm = algorithm
+
+    def check_if_algorithm_exists(self, algorithm):
+        available_algorithms = ["stump"]
```

Comment:
@tcrasset There was a question in [this comment](https://github.com/tslearn-team/tslearn/pull/126#issuecomment-641316807) regarding GPU matrix profiles. STUMPY has a similar function to `stump` called `gpu_stump` which essentially has the same function signature. I recommend adding this as one of the implementations. If no valid NVIDIA GPUs are installed (i.e., no CUDA driver) then STUMPY will check this and error out appropriately before proceeding.

Between `stump` and `gpu_stump`, I think these are probably sufficient